pkg/api/types.go

```diff
@@ -1712,6 +1714,103 @@ type SecretList struct {
 	Items []Secret `json:"items"`
 }
 
+// SecurityContext holds security configuration that will be applied to a container.  If a security context
+// is set on the container spec then it must comply with any constraints defined in the SecurityConstraints context
+// it is running in.  If a security context is not supplied and the pod is running under a SecurityConstraints context
+// then a default SecurityContext may be applied.
+type SecurityContext struct {
+	// Capabilities are the capabilities to add/drop when running the container
+	// TODO: will need to refactor this from the container spec to here
+	Capabilities Capabilities
+
+	// Run the container in privileged mode
+	// TODO: will need to refactor this from the container spec to here
+	Privileged bool
```

Comment:
Hmm, if policy rejections happen up-front (good), then I guess I am fine
with this, though it still feels a little inconsistent as to what goes here
vs elsewhere.

On Tue, Apr 7, 2015 at 4:58 PM, Clayton Coleman <notifications@github.com>
wrote:

> In pkg/api/types.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/6287#discussion_r27932568>
> :
>
> > @@ -1712,6 +1714,103 @@ type SecretList struct {
> >  	Items []Secret `json:"items"`
> >  }
> >
> > +// SecurityContext holds security configuration that will be applied to a container.  If a security context
> > +// is set on the container spec then it must comply with any constraints defined in the SecurityConstraints context
> > +// it is running in.  If a security context is not supplied and the pod is running under a SecurityConstraints context
> > +// then a default SecurityContext may be applied.
> > +type SecurityContext struct {
> > +	// Capabilities are the capabilities to add/drop when running the container
> > +	// TODO: will need to refactor this from the container spec to here
> > +	Capabilities Capabilities
> > +
> > +	// Run the container in privileged mode
> > +	// TODO: will need to refactor this from the container spec to here
> > +	Privileged bool
>
> Technically that would be the policy side - policy rejecting fields not in
> the security context seems reasonable. An example is whether you are
> allowed to pull an image from public registries, or use hostdir
> … <#14c9652a3e456526_>
>  On Apr 7, 2015, at 7:56 PM, Tim Hockin <notifications@github.com> wrote:
> In pkg/api/types.go: > @@ -1712,6 +1714,103 @@ type SecretList struct { >
> Items []Secret `json:"items"` > } > > +// SecurityContext holds security
> configuration that will be applied to a container. If a security context >
> +// is set on the container spec then it must comply with any constraints
> defined in the SecurityConstraints context > +// it is running in. If a
> security context is not supplied and the pod is running under a
> SecurityConstraints context > +// then a default SecurityContext may be
> applied. > +type SecurityContext struct { > + // Capabilities are the
> capabilities to add/drop when running the container > + // TODO: will need
> to refactor this from the container spec to here > + Capabilities
> Capabilities > + > + // Run the container in privileged mode > + // TODO:
> will need to refactor this from the container spec to here > + Privileged
> bool The question is whether ANYONE can create host ports or just with
> special permissions? I really want us to steer people AWAY from bad
> patterns. … — Reply to this email directly or view it on GitHub.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/6287/files#r27932568>
> .
>